### ISSUE TITLE
buildTarget wrong?

### ISSUE THREAD

**Issue by Ray-Luo** [eriklindernoren/PyTorch-YOLOv3#93]
In utils.py from line 223 - 227
```
anch_ious = bbox_iou(gt_box, anchor_shapes)
# Where the overlap is larger than threshold set mask to zero (ignore)
conf_mask[b, anch_ious > ignore_thres, gj, gi] = 0
# Find the best matching anchor box
best_n = np.argmax(anch_ious)
```
line 234 `conf_mask[b, best_n, gj, gi] = 1`

If anch_ious is [0.3, 0.1, 0.6], so best_n is 2. After `conf_mask[b, anch_ious > ignore_thres, gj, gi] = 0` conf_mask will be [1,1,0], but after `conf_mask[b, best_n, gj, gi] = 1` conf_mask will be [1,1,1] again, which should not be?

**Comment by Lausannen** [comment]
Because when best_n is 2,  the mask will be [0,0,1]. And in yolo layer , you can see one line             " "conf_mask_false = conf_mask - mask", finally when we use conf_mask_false , the best_n location will be zero. I think this way is used to make it easier when we calculate the false positive boxes

**Comment by eriklindernoren** [comment]
@Ray-Luo It seems a bit strange I guess, but the paper tells us that we should always train based on the anchor box with the highest iou. For anchors that don't have the highest iou but which still exceeds 0.5 we ignore the confidence loss.
